### PR TITLE
Update cupertino demos in gallery

### LINKED ISSUE #39685: CupertinoDatePicker text color and background color are white in ios 13 dark mode
CupertinoDatePicker text color and background color are white in ios 13 dark mode
Simulator Screen Shot - iPhone Xs - 2019-09-02 at 18 13 49

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
## Description

| light | dark| 
| --- | --- |
| Simulator Screen Shot - iPhone 11 - 2019-10-30 at 11 45 27 | Simulator Screen Shot - iPhone 11 - 2019-10-30 at 11 46 53 |
| Simulator Screen Shot - iPhone 11 - 2019-10-30 at 11 45 44 | Simulator Screen Shot - iPhone 11 - 2019-10-30 at 11 46 57 |
| Simulator Screen Shot - iPhone 11 - 2019-10-30 at 11 45 53 | Simulator Screen Shot - iPhone 11 - 2019-10-30 at 11 47 02 |
| Simulator Screen Shot - iPhone 11 - 2019-10-30 at 11 46 00 |  Simulator Screen Shot - iPhone 11 - 2019-10-30 at 11 47 07 |
| Simulator Screen Shot - iPhone 11 - 2019-10-30 at 11 46 02 |  Simulator Screen Shot - iPhone 11 - 2019-10-30 at 11 47 10 |
| Simulator Screen Shot - iPhone 11 - 2019-10-30 at 11 46 04 | Simulator Screen Shot - iPhone 11 - 2019-10-30 at 11 47 11 |
| Simulator Screen Shot - iPhone 11 - 2019-10-30 at 14 52 47 | Simulator Screen Shot - iPhone 11 - 2019-10-30 at 14 53 43 |
| Simulator Screen Shot - iPhone 11 - 2019-10-30 at 11 46 17 | Simulator Screen Shot - iPhone 11 - 2019-10-30 at 11 47 26 |
| Simulator Screen Shot - iPhone 11 - 2019-10-30 at 14 53 04 |  Simulator Screen Shot - iPhone 11 - 2019-10-30 at 14 53 55 |
| Simulator Screen Shot - iPhone 11 - 2019-10-30 at 11 46 28 | Simulator Screen Shot - iPhone 11 - 2019-10-30 at 11 47 41 |
| Simulator Screen Shot - iPhone 11 - 2019-10-30 at 11 46 32 | Simulator Screen Shot - iPhone 11 - 2019-10-30 at 11 47 44 |
| Simulator Screen Shot - iPhone 11 - 2019-10-30 at 11 46 44 | Simulator Screen Shot - iPhone 11 - 2019-10-30 at 11 47 52|

Can't directly use `CupertinoDynamicColor` in `CupertinoPicker`s because the relevant PR is reverted and has to re-land first.

## Related Issues

https://github.com/flutter/flutter/issues/35541
Fixes #39685

## Tests

I added the following tests:

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by xster** [comment]
Is the black on blue contrast for the cupertino button going to get any better with CupertinoDynamicColor (or is it already using it?).  

The contrast looks a bit low unless the blue is going to look a bit brighter with dynamic colors.

**Comment by LongCatIsLooong** [comment]
@xster thank you for checking! Almost all Cupertino widgets are already using dynamic colors, including `CupertinoButton`. The RGB values of those colors seem correct, but `CupertinoThemeData.primaryContrastingColor`'s default value (white in light mode and black in dark mode, same as the system primary background color) is kinda made up so the primary contrasting color on primary color combination doesn't look as good.  The combo is only used by pre ios 13 segmented controls and `CupertinoButton.filled` buttons. I guess I can make the segmented control's default inactive background color transparent, and change the primary contrasting color to always be white?

**Review by justinmc — APPROVED** [review]
LGTM

The screenshots look really good.  Thanks for doing all the tedious work here!

File: examples/flutter_gallery/lib/demo/cupertino/cupertino_segmented_control_demo.dart [hunk 1]
```diff
@@ -46,7 +46,13 @@ class _CupertinoSegmentedControlDemoState extends State<CupertinoSegmentedContro
     ),
   };
 
-  int sharedValue = 0;
+  final ValueNotifier<int> controller = ValueNotifier<int>(0);
+
+  @override
+  void initState() {
+    super.initState();
+    controller.addListener(() { setState(() {}); });
```
Comment (HansMuller) [inline examples/flutter_gallery/lib/demo/cupertino/cupertino_segmented_control_demo.dart]: I don't think we want to require developers to set up the segmented control this way. The convention for this kind of thing is to for the onValueChanged() callback set update the enclosing widget's state and call setState(). Conventionally, the controller parameter is optional. If the controller is specified then the caller must manage its lifetime.

File: examples/flutter_gallery/lib/demo/cupertino/cupertino_picker_demo.dart [hunk 2]
```diff
@@ -90,7 +94,7 @@ class _CupertinoPickerDemoState extends State<CupertinoPickerDemo> {
               CupertinoPicker(
                 scrollController: scrollController,
                 itemExtent: _kPickerItemHeight,
-                backgroundColor: CupertinoColors.white,
+                backgroundColor: CupertinoDynamicColor.resolve(CupertinoColors.systemBackground, context),
```
[the remark this answers is not in the document]
Comment (LongCatIsLooong) [reply examples/flutter_gallery/lib/demo/cupertino/cupertino_picker_demo.dart]: Currently the `nullOk` parameter of `CupertinoDynamicColor.resolveFrom` defaults to false. I think I should change it to true?

**Comment by LongCatIsLooong** [comment]
`systemBlue` over `white` yields a contrast ratio of 4.01.

**Review by HansMuller — APPROVED** [review]
LGTM

**Pull request merged**